### PR TITLE
[IN-1949] Add Apple cert download

### PR DESCRIPTION
**Pull request by bitrisekristof**
### New Pull Request Checklist

*Please mark the points which you did / accept.*

- [ ] The tool I added is stable, and __does not require frequent updates__. _Preinstalled tools on the LTS stacks
  are not updated, so if the tool requires frequent updates you should handle the installation on-demand (see: [Install Any Additional Tool - DevCenter](https://bitrise-io.github.io/devcenter/tips-and-tricks/install-additional-tools/) for more information)_
- [ ] I updated the corresponding tests if there were any.
- [ ] I added a version report line to [`system_report.sh`](/system_report.sh) for the new tool(s) I added
- [ ] I have added an entry to the CHANGELOG.md with a revisionID (the current date and the number of the change e.g.: 2019_10_11_1) and a brief description of the actual change
- [ ] I have updated the revisionID in [bitrise_profile](roles/profiles/files/bitrise_profile) **BITRISE_OSX_STACK_REV_ID** with the one I just created in the CHANGELOG.md

### CODE REVIEW

File: roles/certs/tasks/main.yml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+---
+
+- name: Apple certificate download
+  shell: bash -l -c "curl https://www.apple.com/certificateauthority/AppleWWDRCAG3.cer --output apple_cert.cer"
```
Comment (peterheja-bitrise) [inline roles/certs/tasks/main.yml:4]: Add absolute path for this, and a conditional, so it does not run always.

File: roles/certs/tasks/main.yml [hunk 2]
```diff
@@ -0,0 +1,8 @@
...
+
+- name: Add Apple certificate
+  shell: bash -l -c "sudo security add-certificates -k /Library/Keychains/System.keychain apple_cert.cer"
```
Comment (peterheja-bitrise) [inline roles/certs/tasks/main.yml:7]: Absolute path, and conditional, so it does not always run.

**Review by peterheja-bitrise — CHANGES_REQUESTED** [review]
We should try to make our desired state idempotent - so there is no change if we run it twice.

File: bitrise.yml [hunk 3]
```diff
@@ -87,7 +87,7 @@ workflows:
         - content: |-
             #!/usr/bin/env bash
             set -e
-            sudo pip3 install "ansible-lint==${ANSIBLE_LINT_VER}" --force-reinstall
+            sudo pip3 install ansible-lint
```
Comment (MrBlaise) [inline bitrise.yml:90]: If you don't pin the linter version it can break after a while. Might be good idea to keep the version pinning.

**Review by fehersanyi-bitrise — APPROVED** [review]

**Review by MrBlaise — APPROVED** [review]

File: roles/certs/tasks/main.yml [hunk 4]
```diff
@@ -0,0 +1,9 @@
+---
+- name: Apple certificate download
+  get_url:
+    url: https://www.apple.com/certificateauthority/AppleWWDRCAG3.cer
+    dest: $HOME/apple_cert.cer
+    checksum: md5:08a45128fa238443623421dd2c9887ab
+
+- name: Add Apple certificate
+  shell: bash -l -c "sudo security add-certificates -k /Library/Keychains/System.keychain $HOME/apple_cert.cer"
```
Comment (MrBlaise) [inline roles/certs/tasks/main.yml:9]: You could register the output of the get_url task and only run this shell command when that has changed using when. This way the command doesn't always run.

**Review by MrBlaise — CHANGES_REQUESTED** [review]

File: roles/certs/tasks/main.yml (hunk 4 above)
Comment (fehersanyi-bitrise) [reply roles/certs/tasks/main.yml:9]: But there will be no case when it is there, it will always have to run.

Comment (MrBlaise) [reply roles/certs/tasks/main.yml:9]: We talked offline

tldr: this is always running on a fresh vm and on failure it is recreated from scratch so I can accept this

**Review by MrBlaise — APPROVED** [review]

File: roles/certs/tasks/main.yml (hunk 4 above)
Comment (peterheja-bitrise) [reply roles/certs/tasks/main.yml:9]: If we ever want to do local development in the near future, then we should do idempotent roles when we are writing them, not later.
That also means we wouldn't be always running against fresh VMs, and also it is nice to describe a desired state, not just series of commands.

What was the problem with the earlier state of this PR?

Comment (fehersanyi-bitrise) [reply roles/certs/tasks/main.yml:9]: it didn't work 🤷‍♂️
also at the current time, this is an absolutely valid state. we don't do local testing, nor local environment. but this certificate NEEDS to be on the machines that use Xcode 12 or later.

**Review by peterheja-bitrise — APPROVED** [review]
.

File: roles/certs/tasks/main.yml (hunk 4 above)
Comment (peterheja-bitrise) [reply roles/certs/tasks/main.yml:9]: Then let's accept this as tech dept for later.
Approved PR, so it can be merged.
